lib/component.js

```diff
@@ -381,6 +385,13 @@ class Component extends WebComponent {
       this.navigate(window.location.hash);
     }
 
+    for (const contextName of this.getConfig(`contexts`)) {
+      const context = this.getContext(contextName);
+      if (context.bindToComponent) {
```

Comment:
I chose to run the callbacks on each repeated connect/disconnect because it seems more intuitive and flexible. I added two additional test cases that specifically tests that callbacks are repeatedly executed when the same component is repeatedly connected/disconnected.